specification/app/resource-manager/Microsoft.App/preview/2024-02-02-preview/LogicAppsExtension.json

```diff
@@ -0,0 +1,697 @@
+            "description": "An unexpected error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List the workflows": {
+            "$ref": "./examples/LogicApps_ListWorkflows.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/containerApps/{containerAppName}/providers/Microsoft.App/logicApps/{logicAppName}/workflows/{workflowName}": {
+      "get": {
+        "tags": [
+          "LogicApps"
+        ],
+        "summary": "Get workflow information by its name",
+        "operationId": "LogicApps_GetWorkflow",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "containerAppName",
+            "in": "path",
+            "description": "Name of the Container App.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[-\\w\\._\\(\\)]+$"
+          },
+          {
+            "name": "logicAppName",
+            "in": "path",
+            "description": "Name of the Logic App, the extension resource.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[-\\w\\._\\(\\)]+$"
+          },
+          {
+            "name": "workflowName",
+            "in": "path",
+            "description": "Workflow name.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-z][a-z0-9]*$",
+            "minLength": 3,
+            "maxLength": 63
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Workflow information returned.",
+            "schema": {
+              "$ref": "#/definitions/WorkflowEnvelope"
+            }
+          },
+          "default": {
+            "description": "An unexpected error response.",
+            "schema": {
+              "$ref": "./CommonDefinitions.json#/definitions/DefaultErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "GET a workflow": {
+            "$ref": "./examples/LogicApps_GetWorkflow.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.App/containerApps/{containerAppName}/providers/Microsoft.App/logicApps/{logicAppName}/deployWorkflowArtifacts": {
+      "post": {
```

Comment:
Would you share more details about using this endpoint to create or update the workflow? And why not add "put" and "delete" methods to the workflow endpoint?

Reply:
The create or update body would look like this
```
{
    "files": {
        "workflow1/workflow.json": {
          <json contents of workflow.json file>
        },
        "workflow2/workflow.json": {
        <json contents of workflow file>
        }
    }
}
```

We designed it this way because we wanted to keep it consistent for UX with the exisiting APIs for logic apps in Microsoft.Web.

Reply:
> keep it consistent for UX with the exisiting APIs for logic apps in Microsoft.Web.

Could you please share the swagger link the for the ` exisiting APIs`

Reply:
Here it is https://github.com/Azure/azure-rest-api-specs/blob/9583ed6c26ce1f10bbea92346e28a46394a784b4/specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/WebApps.json#L22105C5-L22157C8